ts/components/messages/MessageAgenda.tsx

```diff
@@ -266,15 +265,91 @@ const FakeItemComponent = (
 class MessageAgenda extends React.PureComponent<Props, State> {
   // Ref to section list
   private sectionListRef = React.createRef<any>();
+  private idTimeoutProgress?: number;
 
   constructor(props: Props) {
     super(props);
     this.state = {
-      itemLayouts: []
+      itemLayouts: [],
+      firstSectionListUpdate: true,
+      elementVisible: 0,
+      isLoadingProgress: false,
+      isContinuosScrollEnabled: true
     };
     this.loadMoreData = this.loadMoreData.bind(this);
   }
 
+  public componentWillUnmount() {
+    if (this.idTimeoutProgress !== undefined) {
+      clearTimeout(this.idTimeoutProgress);
+    }
+  }
+
+  // This void search if the last deadline is showed
+  private checkIfLastSection() {
```

Comment:
Using Option<string> (see comment below) this function could be simplified as follow

```typescript
// checks if the last section is shown, if yes disables continuos scroll
private checkIfIsLastSection() {
    if (!this.state.isContinuosScrollEnabled) {
      return;
    }
    this.props.lastDeadLineId.foldL(
      () =>
        this.setState({
          isContinuosScrollEnabled: false
        }),
      lastDeadLineId => {
        const isLastDeadLine = this.props.sections
          .map(s => s.data)
          .some(items =>
            items.some(
              item => !isFakeItem(item) && item.e1.id === lastDeadLineId
            )
          );
        if (isLastDeadLine) {
          this.setState({
            isContinuosScrollEnabled: false
          });
        }
      }
    );
  }
```